File: jvm/leetcode_scala/src/main/scala/leetcode/editor/cn/[283]移动零.scala
package leetcode.editor.cn
object MoveZeroes {
//leetcode submit region begin(Prohibit modification and deletion)
object Solution {
    def moveZeroes(nums: Array[Int]): Unit = {
      var slow = 0
      var fast = 0
      while (fast < nums.length) {
        if (nums(fast) != 0){
          nums(slow) = nums(fast)
          slow += 1
        }
        fast += 1
      }

      for (i <- slow until nums.length) nums(i) = 0
    }
}
//leetcode submit region end(Prohibit modification and deletion)

}

File: cpp/leetcodecn/[283]移动零.cpp
#include <vector>
#include "util.h"
#include <map>
#include <string>
#include <utility>
#include <algorithm>
#include <memory>

using namespace std;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
public:
    void moveZeroes(vector<int> &nums) {
        auto slow = nums.begin();
        for (int & num : nums) {
            if (num != 0) {
                *slow = num;
                slow++;
            }
        }

        for_each(slow, nums.end(), [](auto &v) { v = 0; });

    }
};
//leetcode submit region end(Prohibit modification and deletion)


int main(int argv, char *argc[]) {
}